cfme/tests/test_login.py

```diff
@@ -55,3 +56,35 @@ def test_bad_password(context, request, appliance):
         if appliance.version >= '5.9':
             view = appliance.browser.create_view(LoginPage)
             assert view.password.read() == '' and view.username.read() == ''
+
+
+@pytest.mark.parametrize('context', [ViaUI])
+def test_update_password(context, request, appliance):
+    """ Test updating password from the login screen. """
+
+    # First, create a temporary new user
+    username = 'user_temp_{}'.format(fauxfactory.gen_alphanumeric(4).lower())
+    new_creds = Credential(principal=username, secret='redhat')
+    user_group = appliance.collections.groups.instantiate(description="EvmGroup-vm_user")
+    user = appliance.collections.users.create(
+        name=username,
+        credential=new_creds,
+        groups=user_group
+    )
+
+    # Try to login as the new user to verify it has been created
+    logged_in_page = appliance.server.login(user)
+    assert logged_in_page.is_displayed
+    logged_in_page.logout()
+
+    # Now try to login while changing the password for the user
+    changed_pass_page = appliance.server.update_password(new_password='changeme', user=user)
+    assert changed_pass_page.is_displayed
+    changed_pass_page.logout()
+
+    # Try to login with the user with old password
+    error_message = "Incorrect username or password"
+    with error.expected(error_message):
+        appliance.server.login(user)
```

Comment:
**REQUIRED** Where are the credentials being updated on the user's credential object?

Reply:
As far as I understand it, the user object is not being updated. The process of updating a password with "update_password" does change the password in the appliance but not in the user object used in this test.

And it's not entirely a bad thing in this case. When I'm trying to log-in with an old password, I just use the one the user object has stored. This should produce an error, because the password in the appliance is different than we have stored in user object.

Considering that I don't need the user object after the password changes, I had no motivation to go deeper or to update the password for the user manually.

Reply:
OK I understand.